### PR TITLE
e2e Validation for Modeling Rule fields in Schema/TestData

### PR DESCRIPTION
**Pull request by GuyAfik**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-7064

## Description
1. Validate that the schema types are aligned with the TestData types.
2. Add ignore mechanism to ignore errors when testing modeling rules.

### CODE REVIEW

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:302]: use a dict comprehension

**Review by JudahSchwartz — APPROVED** [review]
looks great! Please wait on @daryakoval's approval

File: CHANGELOG.md
Comment (JudahSchwartz) [inline CHANGELOG.md:3]:
```suggestion
* The **modeling-rules test** command will now validate that the modeling rules schema mappings are aligned with the test-data mappings.
```

File: CHANGELOG.md
Comment (daryakoval) [inline CHANGELOG.md]: Why not your addition in green?

File: demisto_sdk/commands/test_content/test_modeling_rule/tests/test_modeling_rule_test.py
Comment (daryakoval) [inline demisto_sdk/commands/test_content/test_modeling_rule/tests/test_modeling_rule_test.py:939]: If your class inherits from str as well, then you will able to use it without `.value`

class Validations(Enum, str):
    SCHEMA_TYPES_ALIGNED_WITH_TEST_DATA = "schema_test_data_types"

```suggestion
                    Validations.SCHEMA_TYPES_ALIGNED_WITH_TEST_DATA
```

File: demisto_sdk/commands/test_content/xsiam_tools/test_data.py
Comment (daryakoval) [inline demisto_sdk/commands/test_content/xsiam_tools/test_data.py:8]:
```suggestion
class Validations(Enum, str):
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (daryakoval) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:721]: If the class inherits from str, you don't need `.value`

```suggestion
            Validations.SCHEMA_TYPES_ALIGNED_WITH_TEST_DATA
```

File: demisto_sdk/commands/test_content/xsiam_tools/test_data.py
Comment (daryakoval) [inline demisto_sdk/commands/test_content/xsiam_tools/test_data.py:8]:
```suggestion
class Validations(Enum, str):
```

File: demisto_sdk/commands/test_content/test_modeling_rule/tests/test_modeling_rule_test.py
Comment (daryakoval) [inline demisto_sdk/commands/test_content/test_modeling_rule/tests/test_modeling_rule_test.py:1518]: Please add a test that catches the case where we have the same field in 2 test data events with different types.

**Review by daryakoval — COMMENTED** [review]
Great work :)

File: CHANGELOG.md
Comment (GuyAfik) [reply CHANGELOG.md]: should be good

**Review by JudahSchwartz — APPROVED** [review]
Great work!

**Pull request merged**
